java/src/Ice/src/main/java/com/zeroc/Ice/_ObjectPrxI.java

```diff
@@ -830,7 +571,7 @@ private void readObject(java.io.ObjectInputStream in)
       }
       _ObjectPrxI proxy = (_ObjectPrxI) communicator.stringToProxy(s);
```

Comment:
No related to the PR, but it would be clear to use the reference factory and _ObjectPrxI constructor and avoid the cast.

Reply:
I went ahead and changed it, as suggested.